trackpy/masks.py

```diff
@@ -30,19 +35,42 @@ def r_squared_mask(radius, ndim):
         coords = np.array(np.meshgrid(*points, indexing="ij"))
     else:
         coords = np.array([points[0]])
-    r = [(coord/rad)**2 for (coord,rad) in zip(coords,radius)]
+    r = [(coord/rad)**2 for (coord, rad) in zip(coords, radius)]
     r2 = np.sum(coords**2, 0).astype(int)
     r2[sum(r) > 1] = 0
     return r2
+    
+
+@memo
+def x_squared_masks(radius, ndim):
+    "Returns ndim masks with values x^2 inside radius and 0 outside"
+    radius = validate_tuple(radius, ndim)
+    points = [np.arange(-rad, rad + 1) for rad in radius]
+    if len(radius) > 1:
+        coords = np.array(np.meshgrid(*points, indexing="ij"))
+    else:
+        coords = np.array([points[0]])
+    r = [(coord/rad)**2 for (coord, rad) in zip(coords, radius)]
+    masks = np.asarray(coords**2, dtype=int)
+    masks[:, sum(r) > 1] = 0
+    return masks
+
+
+@memo
+def root_sum_x_squared(radius, ndim):
+    "Returns the root of the sum of all x^2 inside the mask for each dim."
+    masks = x_squared_masks(radius, ndim)
+    r2 = np.sum(masks, axis=tuple(range(1, ndim + 1)))  # each ax except first
+    return np.sqrt(r2)
 
 
 @memo
 def theta_mask(radius):
     "Mask of values giving angular position relative to center"
     # 2D only
-    tan_of_coord = lambda y, x: np.arctan2(radius - y, x - radius)
-    diameter = 2*radius + 1
-    return np.fromfunction(tan_of_coord, (diameter, diameter))
+    radius = validate_tuple(radius, 2)
+    tan_of_coord = lambda y, x: np.arctan2(radius[0] - y, x - radius[1])
```

Comment:
It is mirrored, so it is 

```
   6
9      3
   12
```

not viewed from the back.  'counter clockwise' is from 3 -> 12.